YourEasyRentTest/Controllers/ProductControllerTest.cs

```diff
@@ -0,0 +1,385 @@
+            var controller = new ProductController(productRepoMock.Object);
+
+            //act
+            var result = await controller.Get(OkProductId);
+
+            //assert
+            result.Should().BeOfType<ActionResult<Product>>()
+                .Which.Result.Should().BeOfType<OkObjectResult>()
+                .Which.Value.Should().BeOfType<Product>();
+
+        }
+        [Fact]
+        public async Task GetProductById_WhenProductByIdNull_ReturnNotFound()
+        {
+            //arrange
+            string NotProductId = "NotProductId";
+            var productRepoMock = new Mock<IProductRepository>();
+            productRepoMock.Setup(repo => repo.Get(It.IsAny<string>())).ReturnsAsync((Product?)null);
+            var controller = new ProductController(productRepoMock.Object);
+
+            //act
+            var result = await controller.Get(NotProductId);
+
+
+            //assert
+            result.Should().BeOfType<ActionResult<Product>>()
+                .Which.Result.Should().BeOfType<NotFoundResult>();
+
+        }
+        [Fact]
+        public async Task GetProductById_WhenExeption_Return500StatusCode()
+        {
+            //arrange
+            string ExceptionId = "ExceptionId";
+            var productRepoMock = new Mock<IProductRepository>();
+            productRepoMock.Setup(repo => repo.Get(It.IsAny<string>())).ThrowsAsync(new Exception("ExceptionId"));
+            var controller = new ProductController(productRepoMock.Object);
+
+            //act
+            var result = await controller.Get(ExceptionId);
+            //assert
+            result.Should().BeOfType<ActionResult<Product>>()
+                .Which.Result.Should().BeOfType<ObjectResult>()
+                .Which.StatusCode.Should().Be(500);
+        }
+
+
+        [Fact]
+        public async Task GetProductByBrand_ProductByBrandFound_ReturnsOkResult()
+        {
+            //arrange
+            string brandWithOkResult = "brandWithOkResult";
+            var products = new List<Product> { new Product { Id = "test", Brand = "BrandProduct" }, new Product { Id = "test", Brand = "BrandProduct" } };
+            var productRepoMock = new Mock<IProductRepository>();
+            productRepoMock.Setup(repo => repo.GetByBrand(brandWithOkResult)).ReturnsAsync(products);
+            var controller = new ProductController(productRepoMock.Object); // Создала экземпляр контроллера ProductsController, передавая фейковый repositoryMock в конструктор.
+
+            // act
+            var result = await controller.GetProductByBrand(brandWithOkResult);
+
+            // assert
+            result.Should().BeOfType<ActionResult<IEnumerable<Product>>>()
+                .Which.Result.Should().BeOfType<OkObjectResult>();
+
+        }
+
+        [Fact]
+        public async Task GetProductByBrand_WhenBrandNull_ReturnEmptyList()
```

Comment:
It should be "WhenNoProducts" WhenBrandNull is misleading, also it's an internal detail.